build.gradle.kts

```diff
@@ -58,90 +58,84 @@ kotlin {
-                kotlin.srcDir("src/nonJvmMain/kotlin")
-            }
-
-            val appleMain32SourceDirs = listOf(
-                "src/appleMain/kotlin"
-            ) + nix32MainSourceDirs
-
-            val appleMain64SourceDirs = listOf(
-                "src/appleMain/kotlin"
-            ) + nix64MainSourceDirs
-
-            val macosX64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val macosX64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val macosArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val macosArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val iosArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val iosArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val iosArm32Main by getting { kotlin.srcDirs(appleMain32SourceDirs) }
-            val iosArm32Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val iosX64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val iosX64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val iosSimulatorArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val iosSimulatorArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val watchosArm32Main by getting { kotlin.srcDirs(appleMain32SourceDirs) }
-            val watchosArm32Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val watchosArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val watchosArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val watchosX64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val watchosX64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val watchosX86Main by getting { kotlin.srcDirs(appleMain32SourceDirs) }
-            val watchosX86Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val watchosSimulatorArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val watchosSimulatorArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val tvosArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val tvosArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val tvosX64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val tvosX64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
-            val tvosSimulatorArm64Main by getting { kotlin.srcDirs(appleMain64SourceDirs) }
-            val tvosSimulatorArm64Test by getting { kotlin.srcDir("src/appleTest/kotlin") }
+        val nonJvmMain by creating { dependsOn(commonMain) }
+        val nonJvmTest by creating { dependsOn(commonTest) }
+        val jsMain by getting { dependsOn(nonJvmMain) }
+        val jsTest by getting { dependsOn(nonJvmTest) }
```

Comment:
i would comment this out, list the tasks, see what the names are like, and update based on that

Reply:
weird, I'm not really sure how it was compiling previously, because `js` and `jvm` targets were only declared if the host was a Mac machine. I added a bunch of `HostManager` checks and it looks like it is building successfully now